### PR TITLE
core: refactor out Block, BlockBase and Header

### LINKED ISSUE #597: Refactor out block and header structures
`Block`, `BlockBase` and `Header` structures are used a lot in different parts of the system (like RPC), but at the moment using them requires importing whole `core` package which is just too much code, so we need to refactor them out.

Technically can be a part of #398 and relates to #510.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #597 .

### CODE REVIEW

File: pkg/core/helper_test.go [hunk 1]
```diff
@@ -71,8 +73,7 @@ func newBlock(index uint32, txs ...*transaction.Transaction) *Block {
 		},
 		Transactions: txs,
 	}
-	_ = b.rebuildMerkleRoot()
-	b.createHash()
```
Comment (volekerb) [inline pkg/core/helper_test.go]: hm. why we don't have to call `createHash()` now?

File: pkg/network/blockqueue_test.go [hunk 2]
```diff
@@ -4,7 +4,8 @@ import (
 	"testing"
 	"time"
 
-	"github.com/CityOfZion/neo-go/pkg/core"
+	"github.com/CityOfZion/neo-go/pkg/core/block"
+
```
Comment (volekerb) [inline pkg/network/blockqueue_test.go]: nit: empty line

File: pkg/core/blockchain.go [hunk 3]
```diff
@@ -10,6 +10,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/CityOfZion/neo-go/pkg/core/mempool"
+
```
Comment (volekerb) [inline pkg/core/blockchain.go]: nit: empty line

File: pkg/core/helper_test.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/core/helper_test.go]: I guess because `Hash()` already does that internally.

**Comment by roman-khimov** [comment]
Apart from other comments, I think that the mempool patch can be improved with types renaming like PoolItem -> Item, PoolItems -> Items, MemPool -> Pool. Looks very good otherwise, although you should also mention the original issue number somewhere in commit messages.

**Pull request merged**